xml2rfc/writers/preptool.py

```diff
@@ -416,7 +416,7 @@ def check_attribute_values(self, e, p):
         for c in e.iter(tags):
             for a in integer_attributes[c.tag]:
                 i = c.get(a)
-                if i and not i.isdigit() and not i==self.get_attribute_defaults(c.tag)[a]:
+                if i and not i.isdigit() and not i==self.get_attribute_defaults(c.tag).get(a):
                     self.err(c, 'Expected <%s> attribute "%s" to be a non-negative integer, but found "%s"' % (c.tag, a, i))
```

Comment:
I think, it's better to use `self.die()` instead of `self.err()` to exit the processing.
With the current change `xml2rfc` crashes again because it can't handle the date conversion.

Reply:
Works for me.
My main objective was to get out the error message before croaking.

Reply:
Actually, this is a generic loop, not just for dates, and it would help to get all error messages out.

Reply:
Yeah, this might result in unwanted failures.
I think it's better to handle this at the date conversion.
I can push a change.